samples/nfc/system_off/src/main.c

```diff
@@ -152,22 +155,16 @@ static void system_off(struct k_work *work)
 }
 
 
-void main(void)
+/**
+ * @brief  Helper function for printing the reason of the last reset.
+ * Can be used to confirm that NCF field actually woke up the system.
+ */
+static void print_reset_reason(void)
 {
 	u32_t reas;
 
-	/* Configure LED-pins */
-	if (dk_leds_init() < 0) {
-		printk("Cannot init LEDs!\n");
-		return;
-	}
-	dk_set_led_on(SYSTEM_ON_LED);
+#if IS_ENABLED(NRF_POWER_HAS_RESETREAS)
```

Comment:
@anangl Could this maybe be abstracted away by nrfx?

Reply:
I don't see any abstraction in nfrx.

Reply:
@doki-nordic Yeah, I'm asking if we should add one.

Reply:
Yes, it looks like a good idea to provide a layer that would handle the common part of those two HALs. I'll let the nrfx guys know.

Reply:
This PR is not something that have to be merged fast. If this abstraction will be done how long we will wait for it? Should we postpone this PR until nrfx changes?

Reply:
@doki-nordic No. I think we will merge this PR without waiting. No need having this hanging.
Once we have the layer we can update.